app/models/ems_refresh/save_inventory_physical_infra.rb

```diff
@@ -37,17 +36,32 @@ def save_ems_physical_infra_inventory(ems, hashes, target = nil)
     ems
   end
 
+  def save_physical_racks_inventory(ems, hashes, target = nil)
+    target = ems if target.nil?
+
+    deletes = target == ems ? :use_association : []
+
+    ems.physical_racks.reset
+    save_inventory_multi(ems.physical_racks, hashes, deletes, [:ems_ref])
+  end
+
   def save_physical_servers_inventory(ems, hashes, target = nil)
     target = ems if target.nil?
 
     ems.physical_servers.reset
-    deletes = if target == ems
-                :use_association
-              else
-                []
-              end
+    deletes = target == ems ? :use_association : []
+
+    child_keys = %i(computer_system asset_detail hosts)
+
+    hashes.each do |h|
+      rack_uid = h.delete(:rack_uid)
+
+      if rack_uid
+        rack = PhysicalRack.find { |rack| rack.ems_ref == rack_uid }
```

Comment:
If you set the whole rack hash to `h[:rack]` in the parser, after it is saved and you update the IDs you can get the rack ID directly from `h[:rack][:id]`, see https://github.com/ManageIQ/manageiq/blob/master/app/models/ems_refresh/save_inventory_infra.rb#L136 for an example of how this is done

Reply:
I guess I finally got that hehe